File: 3.Selection-Sort/selection-sort.js
function find_smallest(array) {
    let smallest = array[0];
    let smallestIndex = 0;

    for (index in array) {
        if (array[index] < smallest) {
            smallest = array[index];
            smallestIndex = index;
        }
    }

    return smallestIndex;
}

function selection_sort(array) {
    const repetitionNumber = array.length;
    let newArray = [];

    for (let i = 0; i < repetitionNumber; i++) {
        const smallest = find_smallest(array);
        newArray.push(array.splice(smallest, 1)[0]);
    }

    return newArray;
}

const myArray = [10, 102, 78, 22, 82, 51];
console.log(selection_sort(myArray));

File: 3.Selection-Sort/selection-sort.py
def find_smallest(array):
    smallest = array[0]
    smallest_index = 0

    for index in range(1, len(array)):
        if array[index] < smallest:
            smallest = array[index]
            smallest_index = index

    return smallest_index


def selection_sort(array):
    new_array = []

    for i in range(0, len(array)):
        smallest = find_smallest(array)
        new_array.append(array.pop(smallest))

    return new_array


my_array = [10, 102, 78, 22, 82, 51]
print(selection_sort(my_array))
